File: python/easy/check_brackets_simple.py
def check_brackets_simple(s):
  stack = []
  for item in s:
    if item == '(':
      stack.append(item)
    else:
      if not stack:
        return False
      else:
        stack.pop()

  if len(stack) == 0:
    return True
  else:
    return False


print(check_brackets_simple('((()))'))
print(check_brackets_simple('((()'))
print(check_brackets_simple('((()))))'))
print(check_brackets_simple('()()()'))
print(check_brackets_simple('()()()())))'))

File: js/easy/checkBracketsSimple.js
function checkBracketsSimple(s) {
  var stack = [];
  for (var i = 0; i < s.length; i++) {
    if (s[i] === '(') {
      stack.push(s[i]);
    } else {
      var last = stack.pop();
      if (!last) return false;
    }
  }
  return stack.length === 0 ? true : false;
}
console.log(checkBracketsSimple('((()))'));
console.log(checkBracketsSimple('((())'));
console.log(checkBracketsSimple('((())))'));
// O(n)
